bazel: a couple c-deps-related fixes

rickystewart opened the pull request:
1. Use a different `lg-page` size for `jemalloc` on M1 Macs. The older
   value resulted in sporadic runtime issues (see https://github.com/cockroachdb/cockroach/issues/81581).
2. Update linked libraries when building `gssapiccl` for `dev`
   scenarios. On Ubuntu the `res_nsearch` function is found in
   `lresolv_wrapper` and the linker can't find it unless we add it to
   the list here.

Closes https://github.com/cockroachdb/cockroach/issues/80389.
Closes https://github.com/cockroachdb/cockroach/issues/81581.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/80594)
<!-- Reviewable:end -->

andreimatei commented:
please keep the PR title and description the same as the commit message for single-commit PRs

rickystewart commented:
@andreimatei I will. This PR is not ready to be reviewed or merged (`do-not-merge`)

rickystewart committed:
bazel: a couple `c-deps`-related fixes

1. Use a different `lg-page` size for `jemalloc` on M1 Macs. The older
   value resulted in sporadic runtime issues (see #81581).
2. Update linked libraries when building `gssapiccl` for `dev`
   scenarios. On Ubuntu the `res_nsearch` function is found in
   `lresolv_wrapper` and the linker can't find it unless we add it to
   the list here.

Closes #80389.
Closes #81581.

Release note: None

mari-crl reviewed (approved):
Nice work! :D
